src/runtime/refiner.ts

```diff
@@ -277,6 +281,16 @@ export class BinaryExpression extends RefinementExpression {
     return null;
   }
 
+  rearrange() {
```

Comment:
Add a type here, it's unclear if this is updating in place or returning the update.